git-town panics if you have additional remotes

### Steps to reproduce

```
git remote add foo <url>
git town sync
```



### Command output

```shell
panic: unknown remote: "foo"

goroutine 1 [running]:
github.com/git-town/git-town/v14/src/git/gitdomain.NewRemote({0xc00002b308, 0xa})
        D:/a/git-town/git-town/src/git/gitdomain/remote.go:14 +0xe5
github.com/git-town/git-town/v14/src/git/gitdomain.NewRemotes({0xc001b2fda0, 0x2, 0xd9ff20?})
        D:/a/git-town/git-town/src/git/gitdomain/remotes.go:13 +0x70
github.com/git-town/git-town/v14/src/git.(*BackendCommands).RemotesUncached(0xc00007d440)
        D:/a/git-town/git-town/src/git/backend_commands.go:364 +0xf0
github.com/git-town/git-town/v14/src/git.(*BackendCommands).Remotes(0xc00007d440)
        D:/a/git-town/git-town/src/git/backend_commands.go:346 +0x25
github.com/git-town/git-town/v14/src/execute.LoadRepoSnapshot({0xc0001dac18, {0x1325c20, 0x0, 0x0}, 0x1, 0x1, 0xc0001bb6e0, {0x0, 0x1, 0x0, ...}, ...})
        D:/a/git-town/git-town/src/execute/load_repo_snapshot.go:49 +0x3af
github.com/git-town/git-town/v14/src/cmd.determineSyncConfig(0x0, 0xc0001bb6e0, 0x0)
        D:/a/git-town/git-town/src/cmd/sync.go:138 +0x1b8
github.com/git-town/git-town/v14/src/cmd.executeSync(0x0, 0x0, 0x0)
        D:/a/git-town/git-town/src/cmd/sync.go:72 +0xae
github.com/git-town/git-town/v14/src/cmd.syncCmd.func1(0xc0001da008, {0x1325c20?, 0x4?, 0xca7d75?})
        D:/a/git-town/git-town/src/cmd/sync.go:51 +0x66
github.com/spf13/cobra.(*Command).execute(0xc0001da008, {0x1325c20, 0x0, 0x0})
        D:/a/git-town/git-town/vendor/github.com/spf13/cobra/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001aa308)
        D:/a/git-town/git-town/vendor/github.com/spf13/cobra/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        D:/a/git-town/git-town/vendor/github.com/spf13/cobra/command.go:1039
github.com/git-town/git-town/v14/src/cmd.Execute()
        D:/a/git-town/git-town/src/cmd/core.go:38 +0x619
main.main()
        D:/a/git-town/git-town/main.go:18 +0x1e

```
```


### Additional information

Git Town 14.1.0

--- a/features/sync/features/other_remote.feature
+++ b/features/sync/features/other_remote.feature
@@ -1,24 +1,20 @@
-Feature: other Git remotes exist
+Feature: ignores other Git remotes
 
   Background:
     Given the current branch is a feature branch "feature"
     And the commits
       | BRANCH  | LOCATION | MESSAGE               |
-      | main    | local    | local main commit     |
-      |         | origin   | origin main commit    |
       | feature | local    | local feature commit  |
       |         | origin   | origin feature commit |
     And a remote "other" pointing to "git@foo.com:bar/baz.git"
     When I run "git-town sync"
 
-  @this
   Scenario: result
     Then it runs the commands
       | BRANCH  | TYPE     | COMMAND                                 |
       | feature | frontend | git fetch --prune --tags                |
       |         | frontend | git checkout main                       |
       | main    | frontend | git rebase origin/main                  |
-      |         | frontend | git push                                |
       |         | frontend | git checkout feature                    |
       | feature | frontend | git merge --no-edit --ff origin/feature |
       |         | frontend | git merge --no-edit --ff main           |
@@ -27,8 +23,9 @@ Feature: other Git remotes exist
   Scenario: undo
     When I run "git-town undo"
     Then it runs the commands
-      | BRANCH  | COMMAND                                                               |
-      | feature | git push --force-with-lease origin {{ sha 'initial commit' }}:feature |
+      | BRANCH  | COMMAND                                                                                |
+      | feature | git reset --hard {{ sha 'local feature commit' }}                                      |
+      |         | git push --force-with-lease origin {{ sha-in-origin 'origin feature commit' }}:feature |
     And the current branch is still "feature"
     And the initial commits exist
     And the initial branches and lineage exist

--- a/features/sync/features/other_remote.feature
+++ b/features/sync/features/other_remote.feature
@@ -1,0 +1,34 @@
+Feature: other Git remotes exist
+
+  Background:
+    Given the current branch is a feature branch "feature"
+    And the commits
+      | BRANCH  | LOCATION | MESSAGE               |
+      | main    | local    | local main commit     |
+      |         | origin   | origin main commit    |
+      | feature | local    | local feature commit  |
+      |         | origin   | origin feature commit |
+    And a remote "other" pointing to "git@foo.com:bar/baz.git"
+    When I run "git-town sync"
+
+  @this
+  Scenario: result
+    Then it runs the commands
+      | BRANCH  | TYPE     | COMMAND                                 |
+      | feature | frontend | git fetch --prune --tags                |
+      |         | frontend | git checkout main                       |
+      | main    | frontend | git rebase origin/main                  |
+      |         | frontend | git push                                |
+      |         | frontend | git checkout feature                    |
+      | feature | frontend | git merge --no-edit --ff origin/feature |
+      |         | frontend | git merge --no-edit --ff main           |
+      |         | frontend | git push                                |
+
+  Scenario: undo
+    When I run "git-town undo"
+    Then it runs the commands
+      | BRANCH  | COMMAND                                                               |
+      | feature | git push --force-with-lease origin {{ sha 'initial commit' }}:feature |
+    And the current branch is still "feature"
+    And the initial commits exist
+    And the initial branches and lineage exist

--- a/src/git/gitdomain/remote.go
+++ b/src/git/gitdomain/remote.go
@@ -1,7 +1,5 @@
 package gitdomain
 
-import "fmt"
-
 // Remote represents a Git remote.
 type Remote string
 
@@ -11,7 +9,7 @@ func NewRemote(id string) Remote {
 			return remote
 		}
 	}
-	panic(fmt.Sprintf("unknown remote: %q", id))
+	return RemoteOther
 }
 
 // Implementation of the fmt.Stringer interface.
@@ -22,11 +20,13 @@ func (self Remote) String() string {
 const (
 	RemoteNone     = Remote("")
 	RemoteOrigin   = Remote("origin")
+	RemoteOther    = Remote("other")
 	RemoteUpstream = Remote("upstream")
 )
 
 var AllRemotes = []Remote{ //nolint:gochecknoglobals
 	RemoteNone,
 	RemoteOrigin,
+	RemoteOther,
 	RemoteUpstream,
 }

--- a/src/git/gitdomain/remote_test.go
+++ b/src/git/gitdomain/remote_test.go
@@ -13,6 +13,7 @@ func TestNewRemote(t *testing.T) {
 		"origin":   gitdomain.RemoteOrigin,
 		"upstream": gitdomain.RemoteUpstream,
 		"":         gitdomain.RemoteNone,
+		"foo":      gitdomain.RemoteOther,
 	}
 	for give, want := range tests {
 		have := gitdomain.NewRemote(give)

--- a/test/cucumber/steps.go
+++ b/test/cucumber/steps.go
@@ -209,8 +209,7 @@ func Steps(suite *godog.Suite, state *ScenarioState) {
 	})
 
 	suite.Step(`^a remote "([^"]+)" pointing to "([^"]+)"`, func(name, url string) error {
-		newRemote := gitdomain.Remote(name)
-		state.fixture.DevRepo.AddRemote(newRemote, url)
+		state.fixture.DevRepo.AddRemote(gitdomain.Remote(name), url)
 		return nil
 	})
 

--- a/test/cucumber/steps.go
+++ b/test/cucumber/steps.go
@@ -208,6 +208,12 @@ func Steps(suite *godog.Suite, state *ScenarioState) {
 		return nil
 	})
 
+	suite.Step(`^a remote "([^"]+)" pointing to "([^"]+)"`, func(name, url string) error {
+		newRemote := gitdomain.Remote(name)
+		state.fixture.DevRepo.AddRemote(newRemote, url)
+		return nil
+	})
+
 	suite.Step(`^branch "([^"]+)" is active in another worktree`, func(branch string) error {
 		state.fixture.AddSecondWorktree(gitdomain.NewLocalBranchName(branch))
 		return nil

